src/controllers/api/account.ts

```diff
@@ -288,54 +287,48 @@ export class AccountController {
 
 		// 6. Check is paymentAccount exists for the customer
 		const accounts = await PaymentAccountService.instance.find({ customer: customerEntity });
-		if (accounts.length === 0) {
-			const key = await new IdentityServiceStrategySetup(customerEntity.customerId).agent.createKey(
-				SupportedKeyTypes.Secp256k1,
-				customerEntity
-			);
-			if (!key) {
-				return response.status(StatusCodes.BAD_REQUEST).json({
-					error: 'PaymentAccount is not found in database: Key was not created',
-				} satisfies UnsuccessfulResponseBody);
-			}
-			paymentAccount = (await PaymentAccountService.instance.create(
-				CheqdNetwork.Testnet,
-				true,
-				customerEntity,
-				key
-			)) as PaymentAccountEntity;
-			if (!paymentAccount) {
-				return response.status(StatusCodes.BAD_REQUEST).json({
-					error: 'PaymentAccount is not found in database: Payment account was not created',
-				} satisfies UnsuccessfulResponseBody);
-			}
-			// Notify
-			await eventTracker.notify({
-				message: EventTracker.compileBasicNotification(
-					'PaymentAccount was not found in database: Payment account with address: ' +
-						paymentAccount.address +
-						' was created'
-				),
-				severity: 'info',
-			});
-		} else {
-			paymentAccount = accounts[0];
+
+		const mainnetAccountResponse = await AccountController.provisionCustomerAccount(
+			CheqdNetwork.Mainnet,
+			accounts,
+			customerEntity
+		);
+		if (!mainnetAccountResponse.success) {
+			return response.status(mainnetAccountResponse.status).json({
+				error: mainnetAccountResponse.error,
+			} satisfies UnsuccessfulResponseBody);
+		}
+
+		const testnetAccountResponse = await AccountController.provisionCustomerAccount(
+			CheqdNetwork.Testnet,
+			accounts,
+			customerEntity
+		);
+		if (!testnetAccountResponse.success) {
+			return response.status(testnetAccountResponse.status).json({
+				error: testnetAccountResponse.error,
+			} satisfies UnsuccessfulResponseBody);
 		}
 
+		const mainnetAccount = mainnetAccountResponse.data;
+		const testnetAccount = testnetAccountResponse.data;
+
 		// 7. Assign default role on LogTo
 		const customDataFromLogTo = await logToHelper.getCustomData(userEntity.logToId);
 
 		// 8. Create custom_data and update the userInfo (send it to the LogTo)
-		if (Object.keys(customDataFromLogTo.data).length === 0 && paymentAccount.address) {
+		if (Object.keys(customDataFromLogTo.data).length === 0 && testnetAccount.address && mainnetAccount.address) {
```

Comment:
if the mainnet account address is undefined we can set it as null in the custom data? useful for staging environment

Reply:
No this will work on staging as well, it's all plan dependant now. You get two accounts (testnet and mainnet) and you operate using them with a plan (test or build)

Reply:
Okay, then I don't think these conditions are necessary. both mainnet and test account addresses are never undefined